perl/cpan/private/carton.bzl

```
@@ -0,0 +1,199 @@
+"""Bazel tools for interfacing with Carton"""
+
+_HUB_BUILD_FILE = """\
+load("@rules_perl//perl:perl_library.bzl", "perl_library")
+
+package(default_visibility = ["//visibility:public"])
+
+DEPENDENCIES = {dependencies}
+
+perl_library(
+    name = "{name}",
+    deps = [
+        "@{name}__" + dep
+        for dep in DEPENDENCIES
+    ],
+    includes = [],
+)
+
+[
+    alias(
+        name = dep,
+        actual = "@{name}__" + dep,
+    )
+    for dep in DEPENDENCIES
+]
+"""
+
+def _perl_cpan_hub_impl(repository_ctx):
+    repository_ctx.file("WORKSPACE.bazel", """workspace(name = "{}")""".format(
+        repository_ctx.name,
+    ))
+
+    repository_ctx.file("BUILD.bazel", _HUB_BUILD_FILE.format(
+        name = repository_ctx.attr.hub_name,
+        dependencies = json.encode_indent(repository_ctx.attr.modules, indent = " " * 4),
+    ))
+
+perl_cpan_hub = repository_rule(
```

again I think I would prefer 
```suggestion
cpan_hub = repository_rule(
```

---

Updated!